### PR TITLE
PHPLIB-494: Fix validation of options with default values

### PR DESCRIPTION
**Pull request by alcaeus**
https://jira.mongodb.org/browse/PHPLIB-494

This fixes a few validation errors where it is possible to use `null` as a valid value for an option, resulting in undefined behaviour. All other options that have a default value are directly checked for their type, without an `isset` check to guard against a missing option since it should always be present due to the array union.

### CODE REVIEW

File: src/Operation/Watch.php [hunk 1]
```diff
@@ -178,7 +178,7 @@ public function __construct(Manager $manager, $databaseName, $collectionName, ar
             'readPreference' => new ReadPreference(ReadPreference::RP_PRIMARY),
         ];
 
-        if (isset($options['fullDocument']) && ! is_string($options['fullDocument'])) {
+        if (! is_string($options['fullDocument'])) {
```
Comment (jmikola) [inline src/Operation/Watch.php]: Do you think we should also check `readPreference` here? Otherwise, a user could pass `null` for ReadPreference and it will [be ignored in Aggregate](https://github.com/mongodb/mongo-php-library/blob/1.5.1/src/Operation/Aggregate.php#L388). In that case, it's possible the client-level read preference (from a connection string) may be applied by libmongoc.

I think my original approach to this issue was going to create a utility function to replace the `$options +=` operation and ensure that `null` values from the user are entirely ignored by unsetting them before we `+=`. That would also circumvent this issue. It would also allow the `isset()` checks to be left as-is; however, I agree that they would serve no purpose since the default options would always be non-null.

Comment (alcaeus) [reply src/Operation/Watch.php]: If a user calls `Collection::watch` or `Database::watch` with a `null` `readPreference` in the options array, the helper method will inject its own read preference into the options array, overriding the default `primary` read preference from the watch operation class.

If a user instantiates the `Watch` operation class directly, they would be able to pass a `null` `readPreference` directly and cause a `null` `readPreference` being passed to the `Aggregate` command. This read preference is only used when resuming the change stream, at which point the `select_server` function will use a primary read preference since it isn't set.

Regardless, I can add a check for completeness to ensure that users don't pass in a `null` `readPreference`.

---

I was thinking about changing all `+=` calls to a method but saw that in those cases, the checks weren't conditional except in the instances fixed here. That's why I decided to update the few occurrences where there was an `isset` instead of changing all array unions.

Comment (jmikola) [reply src/Operation/Watch.php]: Points noted. I agree with adding the check to not allow a `null` `readPreference` option for Watch, though.

Comment (alcaeus) [reply src/Operation/Watch.php]: I've added the check and also added a test to submit a `null` read preference to the Operation
